opal/owner.go

```diff
@@ -123,13 +131,27 @@ func resourceOwnerRead(ctx context.Context, d *schema.ResourceData, m interface{
 		}
 	}
 
-	users, _, err := client.OwnersApi.GetOwnerUsers(ctx, id).Execute()
-	if err != nil {
-		return diagFromErr(ctx, err)
+	ownerHasSourceGroup := false
+	if owner.SourceGroupId.IsSet() {
+		// NOTE: IsSet() is misleading, if the source_group_id value is nil,
+		// IsSet will still return true, we need to check whether there's an
+		// actual source group id in the value to ensure we need to import the
+		// users as well or not
```

Comment:
`nil` is when the value is set but *string points to nil, since it's optional. I think the API may be slightly wrong in this case. I only spotted this via automated testing.